rest_sensors.cpp

```diff
@@ -2171,28 +2171,45 @@ int DeRestPluginPrivate::changeSensorConfig(const ApiRequest &req, ApiResponse &
                 {
                     if (map[pi.key()].type() == QVariant::Double)
                     {
-                        if (sensor->modelId() == QLatin1String("ZHEMI101"))
+                        if (sensor->modelId() == QLatin1String("ZHEMI101") || sensor->modelId().startsWith(QLatin1String("EMIZB-1")))
                         {
                             const uint mode = map[pi.key()].toUInt(&ok);
                             quint16 interfaceMode = 0;
+                            quint8 availableModes = 0;
                             
-                            if (ok && mode == 1)      { interfaceMode = PULSE_COUNTING_ELECTRICITY; }
-                            else if (ok && mode == 2) { interfaceMode = PULSE_COUNTING_GAS; }
-                            else if (ok && mode == 3) { interfaceMode = PULSE_COUNTING_WATER; }
-                            else if (ok && mode == 4) { interfaceMode = KAMSTRUP_KMP; }
-                            else if (ok && mode == 5) { interfaceMode = LINKY; }
-                            else if (ok && mode == 6) { interfaceMode = DLMS_COSEM; }
-                            else if (ok && mode == 7) { interfaceMode = DSMR_23; }
-                            else if (ok && mode == 8) { interfaceMode = DSMR_40; }
-                            else
+                            if(sensor->modelId() == QLatin1String("ZHEMI101"))
```

Comment:
Change request: Spaces after `if` ;-)

The `ok`check could be done first so it doesn't need to be repeated for each value.

```
if (!ok)
{ }
else if (sensor->modelId() == QLatin1String("ZHEMI101"))
{
}
```

Reply:
Sure